src/routers/xcm/Router.sol

```diff
@@ -119,6 +104,16 @@ contract ConnectorXCMRouter {
         emit Deny(user);
     }
 
+    function file(bytes32 what, address gateway_) external auth {
```

Comment:
Do I understand it correctly that this Router would only be operational after 1) deploying it and 2) filing the gateway? Shouldn't we include the gateway address as a constructor param?

Reply:
I guess the reasoning was that either the gateway is deployed first and then filed on the router, or the router is deployed first and filed on the gateway. One of them needs to be fileable and the other in the constructor.